File: DSA_Question/Two_sum/Twosum.java
package Two_sum;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
public class Twosum {
    
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> prevMap = new HashMap<>(); // val:index
        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];
            if (prevMap.containsKey(diff)) {
                return new int[] { prevMap.get(diff), i };
            } else {
                prevMap.put(nums[i], i);
            }
        }
        return new int[] {}; // or throw an exception if no solution is found
    }
}

File: DSA_Question/Two_sum/two_sum.py
class Solution:
    def twoSum(self, nums:List[int], target:int) -> List[int]:
        prevMap = {} # val:index
        for i,v in enumerate(nums):
            diff = target -v
            if diff in prevMap:
                return [prevMap[diff],i]
            else:   
                prevMap[v] =i
        return         